### DISCUSSION TITLE
Error for inline formulas in MDX files

### DISCUSSION THREAD

**Discussion by Loosetooth** [Myriad-Dreamin/typst.ts#627] [Q&A] [answered]
A while back I opened a previous discussion #517 about using rehype-typst with MDX files and Next.js.

I saw the recent new releases for typst.ts, and tried to upgrade [my test project to try out rehype-typst using MDX and Next.js](https://github.com/Loosetooth/next-rehype-typst-mdx) to version `0.5.3`.

It works okay, however, on the [inline formula](https://github.com/Loosetooth/next-rehype-typst-mdx/blob/c227fa6c6edef69d431c44509435feb58d3b2bc3/app/page.mdx?plain=1#L6) an error is thrown when I upgrade `rehype-typst` to the latest `0.5.3` version:

![image](https://github.com/user-attachments/assets/18319ac2-324b-4460-83b3-688bfda62209)

(The error message is shown when hovering over the inline formula which is now in red.)

Any ideas?
You can check out the branch with the `0.5.3` version [here](https://github.com/Loosetooth/next-rehype-typst-mdx/tree/upgrade-to-typst.ts-0.5.3).

EDIT
---
I just realized that apparently I was using a cached version of the `typst-ts-node-compiler` package.
I get the following error when trying to run the project, seemingly complaining about a missing package:
```
$ pnpm dev

> next-rehype-typst-mdx@0.1.0 dev /home/loosetooth/git/next-rehype-typst-mdx
> next dev

node:internal/modules/esm/resolve:215
  const resolvedOption = FSLegacyMainResolve(packageJsonUrlString, packageConfig.main, baseStringified);
                         ^

Error: Cannot find package '/home/loosetooth/git/next-rehype-typst-mdx/node_modules/.pnpm/@myriaddreamin+rehype-typst@0.5.3/node_modules/@myriaddreamin/typst-ts-node-compiler/index.js' imported from /home/loosetooth/git/next-rehype-typst-mdx/node_modules/.pnpm/@myriaddreamin+rehype-typst@0.5.3/node_modules/@myriaddreamin/rehype-typst/lib/index.js
    at legacyMainResolve (node:internal/modules/esm/resolve:215:26)
    at packageResolve (node:internal/modules/esm/resolve:841:14)
    at moduleResolve (node:internal/modules/esm/resolve:927:18)
    at defaultResolve (node:internal/modules/esm/resolve:1169:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:540:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:509:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:239:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:96:40)
    at link (node:internal/modules/esm/module_job:95:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Perhaps when the 0.5.3 version of this package is released and #628 is fixed we can look into the above issue further.

**Comment by Loosetooth** [comment]
Now that the v0.5.3 of `typst-ts-node-compiler` is released correctly, I still get the above error.
When I inspect my `node_modules` folder, I do not have a `typst-ts-node-compiler` folder in it, even though it is a dependency of `rehype-typst`.
When I manually add the dependency to my `package.json` file, a `typst-ts-node-compiler` folder **is** created in the node_modules folder, but it does not contain any `index.js` file:
![image](https://github.com/user-attachments/assets/f80eaac8-767e-46b8-bd4e-12b6d0370f3f)

Any idea what is going on?

**Comment by Myriad-Dreamin** [comment]
The artifact was not bundled correctly, so I published it again as 0.5.4.

**Reply by Loosetooth** [reply, marked as answer]
With 0.5.4 everything seems to work just fine, thanks!

**Comment by Myriad-Dreamin** [comment]
> 不久前，我开启了一场讨论[#517，](https://github.com/Myriad-Dreamin/typst.ts/discussions/517)讨论如何使用 rehype-typst 与 MDX 文件和 Next.js。
> 
> 我看到了 typst.ts 最近发布的新版本，并尝试将[我的测试项目升级到使用 MDX 和 Next.js 的 rehype-typst](https://github.com/Loosetooth/next-rehype-typst-mdx)版本`0.5.3`。
> 
> 它可以正常工作，但是，当我升级到最新版本时[，内联公式](https://github.com/Loosetooth/next-rehype-typst-mdx/blob/c227fa6c6edef69d431c44509435feb58d3b2bc3/app/page.mdx?plain=1#L6)会引发错误：`rehype-typst``0.5.3`
> 
> ![图像](https://private-user-images.githubusercontent.com/66420755/399161563-18319ac2-324b-4460-83b3-688bfda62209.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1NTc4NzMsIm5iZiI6MTczNTU1NzU3MywicGF0aCI6Ii82NjQyMDc1NS8zOTkxNjE1NjMtMTgzMTlhYzItMzI0Yi00NDYwLTgzYjMtNjg4YmZkYTYyMjA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDExMTkzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2NWJkNmZlMWI1MDA2ODMyOWQ1N2MwNWUwYzljZjdmMGQzZjAyOWY3MmU3OGExYjBjYzVmODEwOWQwOTFiZTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Fa9SjZBYwfFPnwplF7lsoMSwFzLLQuS3v_DZSSvbUrA)
> 
> （将鼠标悬停在现在为红色的内联公式上时会显示错误消息。）
> 
> 有什么想法吗？您可以[在此处](https://github.com/Loosetooth/next-rehype-typst-mdx/tree/upgrade-to-typst.ts-0.5.3) 查看带有`0.5.3`版本的分支。[](https://github.com/Loosetooth/next-rehype-typst-mdx/tree/upgrade-to-typst.ts-0.5.3)
> 
> ## 编辑
> 我刚刚意识到我显然正在使用该包的缓存版本`typst-ts-node-compiler`。 尝试运行该项目时，我收到以下错误，似乎在抱怨缺少包：
> 
> ```
> $ pnpm dev
> 
> > next-rehype-typst-mdx@0.1.0 dev /home/loosetooth/git/next-rehype-typst-mdx
> > next dev
> 
> node:internal/modules/esm/resolve:215
>   const resolvedOption = FSLegacyMainResolve(packageJsonUrlString, packageConfig.main, baseStringified);
>                          ^
> 
> Error: Cannot find package '/home/loosetooth/git/next-rehype-typst-mdx/node_modules/.pnpm/@myriaddreamin+rehype-typst@0.5.3/node_modules/@myriaddreamin/typst-ts-node-compiler/index.js' imported from /home/loosetooth/git/next-rehype-typst-mdx/node_modules/.pnpm/@myriaddreamin+rehype-typst@0.5.3/node_modules/@myriaddreamin/rehype-typst/lib/index.js
>     at legacyMainResolve (node:internal/modules/esm/resolve:215:26)
>     at packageResolve (node:internal/modules/esm/resolve:841:14)
>     at moduleResolve (node:internal/modules/esm/resolve:927:18)
>     at defaultResolve (node:internal/modules/esm/resolve:1169:11)
>     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:540:12)
>     at ModuleLoader.resolve (node:internal/modules/esm/loader:509:25)
>     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:239:38)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:96:40)
>     at link (node:internal/modules/esm/module_job:95:36) {
>   code: 'ERR_MODULE_NOT_FOUND'
> }
> ```
> 
> 也许当这个软件包的 0.5.3 版本发布并且#628被修复时，我们可以进一步研究上述问题。

The screenshot seems to have been fixed but I don't know the reason it still happens in `0.5.2`. We might check the version and run it again.
